ote_sdk/ote_sdk/entities/model_template.py

```diff
@@ -75,22 +75,49 @@ def __str__(self) -> str:
 
 class TaskType(Enum):
     """
-    The type of algorithm within the task family.
+    The type of algorithm within the task family. Also contains relevant information
+    about the task type like label domain, if it is trainable, if it is an anomaly task
+    or if it supports global labels.
     """
 
-    NULL = auto()
-    DATASET = auto()
-    CLASSIFICATION = auto()
-    SEGMENTATION = auto()
-    DETECTION = auto()
-    ANOMALY_DETECTION = auto()
-    CROP = auto()
-    TILE = auto()
-    INSTANCE_SEGMENTATION = auto()
-    ACTIVELEARNING = auto()
-    ANOMALY_SEGMENTATION = auto()
-    ANOMALY_CLASSIFICATION = auto()
-    ROTATED_DETECTION = auto()
+    def __new__(cls, *args, **kwds):
+        obj = object.__new__(cls)
+        obj._value_ = args[0]
+        return obj
+
+    def __init__(
+        self,
+        value: int,
+        domain: Domain,
+        is_trainable: bool,
+        is_anomaly: bool,
+        is_global: bool,
+    ):
+        """
+        :param value: Unique integer for .value property of Enum (auto() does not work)
+        :param domain: Label domain of the task type
+        :param is_trainable: Boolean indicating if the task type can be trained
+        :param is_anomaly: Boolean indicating if the task is an anomaly task
+        :param is_global: Boolean indicating if the task support global labels
+        """
+        self.domain = domain
+        self.is_trainable = is_trainable
+        self.is_anomaly = is_anomaly
+        self.is_global = is_global
```

Comment:
I agree that it is a bit strange having these values for all TaskTypes with domain “Domain.NULL”, however for trainable tasks  if it's global it can't be local. I believe in the IMPT repo we also first check if a task is trainable before checking if it is global or local which would work with this approach too. For that reason, I think it is a bit redundant to have is_local.

Reply:
> for trainable tasks if it's global it can't be local

Yeah that's true, but this property exists for both trainable and non-trainable tasks. Since this is an SDK, consumed by SC, OTE and other projects, I think it's especially important to make the interface as clear as possible. It's easy to forget (or even ignore) that `not is_global` in practice does not mean "local" but "local OR non-trainable". With `is_local` we bypass the problem because one will never need to write `not is_global` in his code.

Reply:
Also the concepts of trainability and scope (i.e. locality) are orthogonal. In theory, there could exist tasks implementing non-parametric (or with non-trainable params) algorithms that predict shapes (e.g. Selective Search) or labels. As we are at the lowest possible level, the SDK, we must make as few assumptions as possible.

Reply:
Good point. Added